### PR TITLE
(vitineth/typescript_eslint_update) Integrating eslint and updating typescript configuration

### PR DESCRIPTION
**Pull request by Vitineth**
Typescript now builds to a different folder and eslint will lint everything and is available under the `npm run lint` command

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                selector: 'variableLike',
+                format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
+            },
+        ],
+        '@typescript-eslint/indent': ['error', 4],
+        'import/first': ['off'],
+        'import/order': ['off'],
+        'import/prefer-default-export': ['off'],
+        'max-len': ['error', 120],
+        'object-curly-newline': ['off'],
```
Comment (Vitineth) [inline .eslintrc.js:32]: Configuration has been changed slightly to make this fit the project slightly better. These changes may be needed across the project eventually but are fine for the time being

Comment (Lan2u) [reply .eslintrc.js:32]: Are we set on using the airbnb style guide?
I think its a good move but just to confirm.

Comment (Vitineth) [reply .eslintrc.js:32]: Not set on it, I tend to just use it as a base to work from and then tweak it. Haven't looked too much into the other options but Google and Standard are the other two popular ones I think?

Comment (Lan2u) [reply .eslintrc.js:32]: I'm fine just going with the airbnb style for now, we can change as needed later.

**Pull request merged**